contributors/design-proposals/cri-dockershim-checkpoint.md

```diff
@@ -0,0 +1,125 @@
+# CRI: Dockershim PodSandbox Checkpoint 
+
+## Umbrella Issue
+[#34672](https://github.com/kubernetes/kubernetes/issues/34672)
+
+## Background
+[Container Runtime Interface (CRI)](../devel/container-runtime-interface.md) 
+is an ongoing project to allow container runtimes to integrate with 
+kubernetes via a newly-defined API. 
+[Dockershim](https://github.com/kubernetes/kubernetes/blob/release-1.5/pkg/kubelet/dockershim) 
+is the Docker CRI implementation. This proposal aims to introduce 
+checkpoint mechanism in dockershim.
+
+## Motivation
+### Why do we need checkpoint?
+
+
+With CRI, Kubelet only passes configurations (SandboxConfig, 
+ContainerConfig and ImageSpec) when creating sandbox, container and 
+image, and only use the reference id to manage them after creation. 
+However, information in configuration is not only needed during creation. 
+
+In the case of dockershim with CNI network plugin, CNI plugins needs 
+the same information from PodSandboxConfig at creation and deletion. 
+
+```
+Kubelet ---------------------------------
+     | RunPodSandbox(PodSandboxConfig)
+     | StopPodSandbox(PodSandboxID)
+     V
+Dockershim-------------------------------
+     | SetUpPod
+     | TearDownPod
+     V
+Network Plugin---------------------------
+     | ADD
+     | DEL
+     V
+CNI plugin-------------------------------
+```
+
+
+In addition, checkpoint helps to improve the reliability of dockershim. 
+With checkpoints, critical information for disaster recovery could be 
+preserved. Kubelet makes decisions based on the reported pod states 
+from runtime shims. Dockershim currently gathers states from docker 
+engine. However, in the case of disaster recovery, docker engine may 
```

Comment:
Can you help me understand the "disaster recovery" scenario here? Clearly if the machine gets blown away this isn't an issue - is it just if the Docker engine dies?